src/controller/tests/TestAutoCommissioner.cpp

```diff
@@ -0,0 +1,253 @@
+{
+    app::Clusters::TimeSynchronization::Structs::DSTOffsetStruct::Type sDSTBuf;
+
+    sDSTBuf.offset        = int32_t{ 10 };
+    sDSTBuf.validStarting = uint64_t{ 1742998643 };
+    sDSTBuf.validUntil    = uint64_t{ 1842998643 };
+    app::DataModel::List<app::Clusters::TimeSynchronization::Structs::DSTOffsetStruct::Type> list(&sDSTBuf, 1);
+
+    mParams.SetDSTOffsets(list);
+
+    auto r = mCommissioner.SetCommissioningParameters(mParams);
+
+    auto commissioning_params = mCommissioner.GetCommissioningParameters();
+
+    ASSERT_EQ(r, CHIP_NO_ERROR);
+    ASSERT_TRUE(commissioning_params.GetDSTOffsets().HasValue());
+    ASSERT_EQ(commissioning_params.GetDSTOffsets().Value().size(), size_t{ 1 });
+    ASSERT_EQ(commissioning_params.GetDSTOffsets().Value()[0].offset, 10);
+    ASSERT_EQ(commissioning_params.GetDSTOffsets().Value()[0].validStarting, uint64_t{ 1742998643 });
+    ASSERT_EQ(commissioning_params.GetDSTOffsets().Value()[0].validUntil, uint64_t{ 1842998643 });
+}
+
+TEST_F(AutoCommissionerTest, FeaturesPassedTimeZoneValue)
+{
+    app::Clusters::TimeSynchronization::Structs::TimeZoneStruct::Type sTimeZoneBuf;
+
+    char country_name[4];
+    country_name[0] = 'A';
+    country_name[1] = 'R';
+    country_name[2] = 'G';
+    country_name[3] = 0;
+
+    sTimeZoneBuf.offset  = int32_t{ 10 };
+    sTimeZoneBuf.validAt = uint64_t{ 1842998643 };
+    sTimeZoneBuf.name.SetValue(chip::CharSpan{ country_name, 4 });
+
+    app::DataModel::List<app::Clusters::TimeSynchronization::Structs::TimeZoneStruct::Type> list(&sTimeZoneBuf, 1);
+    mParams.SetTimeZone(list);
+
+    auto r = mCommissioner.SetCommissioningParameters(mParams);
+
+    auto commissioning_params = mCommissioner.GetCommissioningParameters();
+
+    ASSERT_EQ(r, CHIP_NO_ERROR);
+    ASSERT_TRUE(commissioning_params.GetTimeZone().HasValue());
+    ASSERT_EQ(commissioning_params.GetTimeZone().Value().size(), size_t{ 1 });
+    ASSERT_EQ(commissioning_params.GetTimeZone().Value()[0].offset, 10);
+    ASSERT_EQ(commissioning_params.GetTimeZone().Value()[0].validAt, uint64_t{ 1842998643 });
+    ASSERT_TRUE(commissioning_params.GetTimeZone().Value()[0].name.HasValue());
+    ASSERT_EQ(commissioning_params.GetTimeZone().Value()[0].name.Value()[0], 'A');
+    ASSERT_EQ(commissioning_params.GetTimeZone().Value()[0].name.Value()[1], 'R');
+    ASSERT_EQ(commissioning_params.GetTimeZone().Value()[0].name.Value()[2], 'G');
+    ASSERT_EQ(commissioning_params.GetTimeZone().Value()[0].name.Value()[3], 0);
+}
+
+TEST_F(AutoCommissionerTest, FeaturesPassedNTPValue)
+{
+    char default_ntp_buffer[20];
+    default_ntp_buffer[0] = 'd';
+    default_ntp_buffer[1] = 'e';
+    default_ntp_buffer[2] = 'f';
+    default_ntp_buffer[3] = 'a';
+    default_ntp_buffer[4] = 'u';
+    default_ntp_buffer[5] = 'l';
+    default_ntp_buffer[6] = 't';
+    default_ntp_buffer[7] = 0;
+
+    mParams.SetDefaultNTP(chip::app::DataModel::MakeNullable(CharSpan{ default_ntp_buffer, 20 }));
```

Comment:
this sends uninitialized data to the method (we claim a 20-byte buffer, but initialize the front only)